### PR TITLE
(#80) Removed all usages of org.hamcrest.MatcherAssert class

### PR DESCRIPTION
**Pull request by borysfan**
This PR is for PDD #80:
- it removes all usages of org.hamcrest.MatcherAssert class in tests
- it prohibits usage of org.hamcrest.MatcherAssert

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#97](https://github.com/llorllale/cactoos-matchers/issues/97) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#97](https://github.com/llorllale/cactoos-matchers/issues/97) is assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx), here is [why](https://www.0crat.com/footprint/C63314D6Z/8b88c019-f46d-4ac8-bd2b-af84b8124339); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/test/java/org/llorllale/cactoos/matchers/ScalarHasValueTest.java [hunk 1]
```diff
@@ -56,46 +55,46 @@
     @Test
     public void matchesAsExpectedWithString() {
         final String expected = "some text";
-        MatcherAssert.assertThat(
-            "doesn't match a String",
-            new UncheckedScalar<>(() -> expected),
+        new Assertion<>(
+            "must match with expected string",
```
Comment (fabriciofx) [inline src/test/java/org/llorllale/cactoos/matchers/ScalarHasValueTest.java]: @borysfan Please, start with a capital letter.

File: src/test/java/org/llorllale/cactoos/matchers/ScalarHasValueTest.java [hunk 2]
```diff
@@ -56,46 +55,46 @@
...
     }
 
     @Test
     public void matchesAsExpectedWithMatcher() {
         final String expected = "text";
-        MatcherAssert.assertThat(
-            "doesn't match a Matcher",
-            new UncheckedScalar<>(() -> expected),
+        new Assertion<>(
+            "must match a Matcher",
```
Comment (fabriciofx) [inline src/test/java/org/llorllale/cactoos/matchers/ScalarHasValueTest.java]: @borysfan Please, start with a capital letter.

File: src/test/java/org/llorllale/cactoos/matchers/ScalarHasValueTest.java [hunk 3]
```diff
@@ -56,46 +55,46 @@
     @Test
     public void matchesAsExpectedWithString() {
```
Comment (fabriciofx) [inline src/test/java/org/llorllale/cactoos/matchers/ScalarHasValueTest.java]: @borysfan `ExpectedWithString` or `WithExpectedString`?

File: src/test/java/org/llorllale/cactoos/matchers/ScalarHasValueTest.java [hunk 4]
```diff
@@ -29,7 +29,6 @@
 
 import org.cactoos.scalar.Constant;
```
Comment (fabriciofx) [inline src/test/java/org/llorllale/cactoos/matchers/ScalarHasValueTest.java]: @borysfan Please, check this file too because it seems has a PDD too `@todo #81:30min Replace all uses of MatcherAssert.assertThat() with...`

File: src/test/java/org/llorllale/cactoos/matchers/ScalarHasValueTest.java [hunk 5]
```diff
@@ -56,46 +55,46 @@
...
     @Test
-    public void hasMatcherDescriptionForFailedTest() throws Exception {
+    public void hasMatcherDescriptionForFailedTest() {
         this.exception.expect(AssertionError.class);
         this.exception.expectMessage(
             new StringContains("Expected: Scalar with \"something\"")
         );
-        MatcherAssert.assertThat(
+        new Assertion<>(
             "missing matcher description",
```
Comment (fabriciofx) [inline src/test/java/org/llorllale/cactoos/matchers/ScalarHasValueTest.java]: @borysfan Please, start with a capital letter.

File: src/test/java/org/llorllale/cactoos/matchers/ScalarHasValueTest.java [hunk 6]
```diff
@@ -56,46 +55,46 @@
...
+        ).affirm();
     }
 
     @Test
-    public void hasMismatchDescriptionForFailedTest() throws Exception {
+    public void hasMismatchDescriptionForFailedTest() {
         this.exception.expect(AssertionError.class);
         this.exception.expectMessage(
-            new StringContains("but: was \"actual\"")
+            new StringContains("but was: was \"actual\"")
```
Comment (fabriciofx) [inline src/test/java/org/llorllale/cactoos/matchers/ScalarHasValueTest.java]: @borysfan There is two "was". Is it correct?

File: src/test/java/org/llorllale/cactoos/matchers/ScalarHasValueTest.java [hunk 7]
```diff
@@ -56,46 +55,46 @@
...
         );
-        MatcherAssert.assertThat(
+        new Assertion<>(
             "missing mismatch description",
```
Comment (fabriciofx) [inline src/test/java/org/llorllale/cactoos/matchers/ScalarHasValueTest.java]: @borysfan Please, start with a capital letter

File: src/test/java/org/llorllale/cactoos/matchers/ScalarHasValueTest.java (hunk 4 above)
Comment (borysfan) [reply src/test/java/org/llorllale/cactoos/matchers/ScalarHasValueTest.java]: @fabriciofx I covered this PDD in my changes -> PDD removed.

File: src/test/java/org/llorllale/cactoos/matchers/ScalarHasValueTest.java (hunk 7 above)
Comment (borysfan) [reply src/test/java/org/llorllale/cactoos/matchers/ScalarHasValueTest.java]: I removed duplicated "was" from ScalarHasValue class.

**Comment by borysfan** [comment]
@fabriciofx I made changes according to your comments.

**Comment by fabriciofx** [comment]
@borysfan This PR seems fine to me.

**Comment by fabriciofx** [comment]
@rultor merge

File: src/main/java/org/llorllale/cactoos/matchers/ScalarHasValue.java [hunk 8]
```diff
@@ -82,9 +82,6 @@ protected boolean matchesSafely(final Scalar<T> item) {
     protected void describeMismatchSafely(final Scalar<T> item,
         final Description description) {
         description
-            .appendText("was ")
-            .appendValue(
-                new UncheckedScalar<>(item).value()
-            );
+            .appendValue(new UncheckedScalar<>(item).value());
```
Comment (llorllale) [inline src/main/java/org/llorllale/cactoos/matchers/ScalarHasValue.java]: @borysfan why this change?

Comment (borysfan) [reply src/main/java/org/llorllale/cactoos/matchers/ScalarHasValue.java]: Please take look at test case hasMismatchDescriptionForFailedTest (https://github.com/llorllale/cactoos-matchers/pull/97/files#diff-b1851e044751c294fbcb4a9467a5f931R86) Before that change on mismatch situation the description was: 'but was: was "something"'. @fabriciofx suggested that we should remove this duplication.

Comment (fabriciofx) [reply src/main/java/org/llorllale/cactoos/matchers/ScalarHasValue.java]: @llorllale I've thought `was: was` strange... :)

Comment (llorllale) [reply src/main/java/org/llorllale/cactoos/matchers/ScalarHasValue.java]: @borysfan @fabriciofx let's keep PRs tightly focused please! :)

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
The job [#97](https://github.com/llorllale/cactoos-matchers/issues/97) is now out of scope

**Comment by 0crat** [comment]
Order was finished: +15 point(s) just awarded to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx)

**Comment by 0crat** [comment]
Job `gh:llorllale/cactoos-matchers#97` doesn't exist in WBS, can't create order

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)
